console/program/src/data_types/value_type/mod.rs

```diff
@@ -38,6 +40,20 @@ pub enum ValueType<N: Network> {
     Future(Locator<N>),
 }
 
+impl<N: Network> ValueType<N> {
+    /// Returns the variant of the value type.
+    pub const fn variant(&self) -> Variant {
```

Comment:
nit: the same could be achieved with `EnumIndex`, which is already derived for `ValueType`

Reply:
I added a `Variant` for consistency and to be more explicit in matching the `Variant` between types. I felt it was a bit vague to use EnumIndex.